core/src/app.rs

```diff
@@ -426,41 +426,43 @@ impl App {
             // Value::None?
             if block.block_type() == BlockType::Reduce {
                 assert!(current_map.is_some());
-                envs =
-                    envs.iter()
-                        .map(|map_envs| {
-                            assert!(map_envs.len() > 0);
-                            let mut env = map_envs[0].clone();
-                            current_map_blocks
-                                .iter()
-                                .map(|n| {
-                                    env.state.insert(
-                                        n.clone(),
-                                        Value::Array(
-                                            map_envs
-                                                .iter()
-                                                .map(|e| match e.state.get(n) {
-                                                    None => Err(anyhow!(
-                                                    "Missing block `{}` output, at iteration {}",
-                                                    n, e.map.as_ref().unwrap().iteration
-                                                ))?,
-                                                    Some(v) => Ok(v.clone()),
-                                                })
-                                                .collect::<Result<Vec<_>>>()?,
-                                        ),
-                                    );
-                                    Ok(())
-                                })
-                                .collect::<Result<Vec<_>>>()?;
-                            env.map = None;
-                            Ok(vec![env])
-                        })
-                        .collect::<Result<Vec<_>>>()?;
+                if envs.iter().any(|map_envs| map_envs.is_empty()) {
+                    envs = vec![];
+                } else {
+                    envs =
```

Comment:
Indentation is likely broken here?

Reply:
Seems alright on my IDE 🤔

Reply:
:+1: the linter is the ground truth anyway. Maybe I just don't see the removal of spaces